src/test/java/core/basesyntax/service/fruitservice/FruitRecordStrategyImplTest.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax.service.fruitservice;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.handlers.AddOperationStrategy;
+import core.basesyntax.service.handlers.RecordHandler;
+import core.basesyntax.service.handlers.RemoveOperationStrategy;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitRecordStrategyImplTest {
+    private static RecordHandler addOperation;
+    private static FruitRecordStrategy fruitRecordStrategy;
+    private static RecordHandler removeOperation;
+    private RecordHandler actual;
+
+    @BeforeClass
+    public static void setUp() {
+        FruitDao fruitDao = new FruitDaoImpl();
+        addOperation = new AddOperationStrategy(fruitDao);
+        removeOperation = new RemoveOperationStrategy(fruitDao);
+        Map<Operation, RecordHandler> fruitOperationHandler = new HashMap<>();
+        fruitOperationHandler.put(Operation.getOperationByLetter("s"), addOperation);
+        fruitOperationHandler.put(Operation.getOperationByLetter("r"), addOperation);
+        fruitOperationHandler.put(Operation.getOperationByLetter("b"), addOperation);
+        fruitOperationHandler.put(Operation.getOperationByLetter("p"), removeOperation);
+        fruitRecordStrategy = new FruitRecordStrategyImpl(fruitOperationHandler);
+    }
+
+    @Test
+    public void check_getOperation_OK() {
+        actual = fruitRecordStrategy.get(Operation.BALANCE);
+        Assert.assertEquals(addOperation, actual);
+        actual = fruitRecordStrategy.get(Operation.RETURN);
+        Assert.assertEquals(addOperation, actual);
+        actual = fruitRecordStrategy.get(Operation.SUPPLY);
+        Assert.assertEquals(addOperation, actual);
+
+        actual = fruitRecordStrategy.get(Operation.PURCHASE);
+        Assert.assertEquals(removeOperation, actual);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void check_getOperation_Not_OK() {
+        actual = fruitRecordStrategy.get(Operation.getOperationByLetter(null));
+        Assert.assertEquals(addOperation, actual);
+
+        actual = fruitRecordStrategy.get(Operation.getOperationByLetter("x"));
```

Comment:
you should split this test into 3 because code won't execute after first exception is thrown, so you won't know if other cases work as you expect. 
And if you wqnt to test exception you don't need assertEquals here